chain/chain/src/chain.rs

```diff
@@ -4623,8 +4636,9 @@ impl<'a> ChainUpdate<'a> {
                     if let Some(chain_flat_storage) =
                         self.runtime_adapter.get_flat_storage_state_for_shard(shard_id)
                     {
-                        let delta =
-                            FlatStateDelta::from_wrapped_trie_changes(&apply_result.trie_changes);
+                        let delta = FlatStateDelta::from_state_changes(
+                            &apply_result.trie_changes.state_changes(),
+                        );
                         chain_flat_storage.add_delta(&block_hash, delta);
```

Comment:
Could you move the part for writing state changes to disk to here? it's possible that we may decide deltas should be stored or added in another place in the future, but it doesn't make sense that `add_delta` which adds the delta to memory and the logic of writing delta to disk are in different places. We should probably add another field in chain_store for the deltas for them to be committed at the same time as the block. For the purpose of this block, you can simply just commit the store_update in here and add a TODO.